File: forest/thornbush-farm.lua
loop
    self:moveXY(43, 50)
    local e = self:findNearestEnemy()
    if e then
        self:buildXY("fire-trap", 43, 50)
    end

    self:moveXY(25, 34)
    e = self:findNearestEnemy()
    if e then
        self:buildXY("fire-trap", 25, 34)
    end

    self:moveXY(43, 20)
    e = self:findNearestEnemy()
    if e then
        self:buildXY("fire-trap", 43, 20)
    end
end

File: forest/thornbush-farm.clj
(defn patrol [x y]
  (.moveXY this x y)
  (def e (.findNearest this (.findEnemies this)))
  (if e
    (if (= (.type e) "ogre")
      (.buildXY this "fire-trap" x y))))

(while true
  (patrol 43 50)
  (patrol 25 34)
  (patrol 43 20))
